dev/tasks/build.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* jshint node: true */
+
+'use strict';
+
+module.exports = function( grunt ) {
+	grunt.registerTask( 'build', 'Build a release out of the current development code.', function() {
+		var done = this.async();
+		module.exports.build( done );
+	} );
+};
+
+// Exports the build method so it can be used from plain node code as well.
+module.exports.build = function( done ) {
+	var target = 'build';
+	var tmp = 'tmp';
+	var stepCounter = 0;
+
+	var tasks = [
+		[ cleanup, 'Cleaning the "' + target + '" directory...' ],
+		[ copyToTmp, 'Copying source files for manipulation...' ],
+		[ removeAmdNamespace, 'AMD cleanup...' ],
+		[ optimize, 'Creating the optimized code...' ],
+		[ cleanupTmp, 'Removing the "' + tmp + '" directory...' ]
+	];
+
+	runNext();
+
+	function runNext() {
+		var next = tasks.shift();
+
+		if ( next ) {
+			stepCounter++;
+			console.log( stepCounter + '. ' + next[ 1 ] );
+			next[ 0 ]( runNext );
+		} else {
+			if ( done ) {
+				done();
+			}
+		}
+	}
+
+	function cleanup( callback ) {
+		var del = require( 'del' );
+		del.sync( target );
```

Comment:
Just to make the code simpler, as we have another call right after that. In any case we need to wait for you anyway, so it should not be a big del, right?